### PR TITLE
Makefile updates

### PR DESCRIPTION
**Pull request by jrincayc**
Combining #45  and re-adding some missing features.

### CODE REVIEW

**Comment by jrincayc** [comment]
It now builds the ucblogo.dmg file correctly on macOS.

@barak Have I broken anything that you need?  (I made ucblogo.pdf part of the dist-gzip files, which possibly might cause problems)

@brianharvey FYI: this removes configure from the git repository, so automake is now needed to build from the git sources. I am debating re-adding this, but the makefile now has a dist-gzip target that generates a tar.gz with configure generated already, so for releases this is not a problem.

I plan on doing a re-review of the changes, and then committing, probably next week.

**Comment by barak** [comment]
Just checked, and "make dist" includes all the files in git, augmented by some generated files. And it works for Debian.

I think the "right thing" is to not include any generated files in git; those go in the "make dist" generated tarball, for anyone that need a pre-generated ./configure etc.

tldr: looks great to me
